### PR TITLE
Add make notes target

### PR DESCRIPTION
**Pull request by miekg**
Make writing the release notes easier with a 'notes' target. This
outputs the authors and a list of PRs in a markdown compatible format.
This can be streamed or copied directly into the release notes.

Signed-off-by: Miek Gieben <miek@miek.nl>

### CODE REVIEW

File: Makefile.release [hunk 1]
```diff
@@ -1,21 +1,13 @@
 # Makefile for releasing CoreDNS
 #
 # The release is controlled from coremain/version.go. The version found there is
-# used to tag the git repo and to build the assets that are uploaded to github
-# (after some sanity checks).
+# used to tag the git repo and to build the assets that are uploaded to GitHub.
 #
 # The release should be accompanied by release notes in the notes/ subdirectory.
 # These are published on coredns.io. For example see: notes/coredns-1.5.1.md
+# Use make -f Makefile.release DOCKER=bla notes to create a skeleton notes document.
```
Comment (johnbelamaric) [inline Makefile.release]: do you need DOCKER=bla in the make command since it's hardcoded in the target?

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [coredns/coredns] Add make note..." ]
>@johnbelamaric commented on this pull request.
>
>━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
>
>In Makefile.release:
>
>>  #
> # The release should be accompanied by release notes in the notes/ subdirectory.
> # These are published on coredns.io. For example see: notes/coredns-1.5.1.md
>+# Use make -f Makefile.release DOCKER=bla notes to create a skeleton notes document.
>
>do you need DOCKER=bla in the make command since it's hardcoded in the target?

OK, I didn't know this, but you _can_ do this checks in the rules. Changed the makefile to
do this and amended the docs.

/Miek

--
Miek Gieben

File: Makefile.release [hunk 2]
```diff
@@ -1,21 +1,13 @@
...
-# Getting the authors for this release is done with the following command line
-# See the 'authors' target.
-#
-# git log --pretty=format:'%an' $(git describe --tags --abbrev=0)..master | sort -u
-#
-# Getting all pull requests merged since the last tag can be done with this "oneliner"
-# See the 'prs' target.
-#
-# % git log $(git describe --tags --abbrev=0)..HEAD --oneline | awk ' { $1="";print } ' | sed 's/^ //' | sed -e 's|#\([0-9]\)|https://github.com/coredns/coredns/pull/\1|'
+# Be sure to prune the PR list a bit, not everything is not worthy!
```
Comment (johnbelamaric) [inline Makefile.release]: s/not worthy/note worthy/

**Review by johnbelamaric — COMMENTED** [review]
one typo

File: Makefile.release (hunk 2 above)
Comment (miekg) [reply Makefile.release]: done

**Pull request merged**